programs/zstdcli.c

```diff
@@ -672,7 +675,25 @@ int main(int argCount, const char* argv[])
         }
     }
 #endif
-
+    if (operation == zom_list) {
+        g_displayOut = stdout;
+        if(filenameIdx==0){
+            DISPLAY("No files given\n");
+            CLEAN_RETURN(0);
+        }
+        DISPLAY("===========================================\n");
+        DISPLAY("Printing information about compressed files\n");
+        DISPLAY("===========================================\n");
+        DISPLAY("Number of files listed: %d\n", filenameIdx);
+        {
```

Comment:
As software architecture guideline,
try to limit the amount of "intelligent" work done inside `zstdcli.c`.
`zstdcli` should be just about managing input, and redirecting them to appropriate module.

Here, you could have a function `FIO_listMultipleFiles()` that would do the loop and printing inside `fileio.c`.
It's not much, but it puts the workload in the expected location.